refactor(tools)!: docs and demos refactor

bennypowers opened the pull request:
## What I Did
- refactor the dev server demos
  - use koa router
  - use `/components/jazz-hands` urls
- refactor the demo page builds in 11ty
  - allow multiple demos for elements
  - simplify demo rendering code
  - generate demos entries in custom elements manifests

zeroedin reviewed (commented):
This looks great.  I did my best to go through line by line and follow the changes.  I believe the breaking change of `/demo` to `/components` is a small trade-off for what we gain here.  Using the custom-elements-manifest plugin for the demo info is a nice place to document that metadata.  I now have a better understanding of how 11ty uses the custom-elements-manifest to populate content.

bennypowers commented:
@zeroedin ok please approve if you think it's ready to roll

brianferry commented:
@bennypowers - Made a small change to the PR with an update to fix 

![image](https://user-images.githubusercontent.com/9708152/180027033-40ec7f0e-1f38-441b-9ca9-faa287c20ae0.png)

bennypowers commented:
Nice catch @brianferry. If you're GTG please drop an approval and we'll get this show on the road

zeroedin commented:
@brianferry and I found some issues based on browser usage:

When you load this PR's changes downstream in the RHDS repo:

In Firefox `rh-alert` generates the following console log:
```
Uncaught TypeError: Error resolving module specifier “@patternfly/pfe-band”. Relative module specifiers must start with “./”, “../” or “/”. rh-alert.js:1:7
^^ Module TypeError above is polyfilled and can be ignored ^^ es-module-shims.js:34:12
```

Safari seems to sometimes load the polyfill (maybe?) for some components such as `rh-alert` but not for `rh-dialog`
```
[Error] TypeError: Module specifier, '@patternfly/pfe-band' does not start with "/", "./", or "../". Referenced from http://localhost:8080/components/alert/demo/rh-alert.js
	promiseReactionJob
[Log] ^^ Module TypeError above is polyfilled and can be ignored ^^ (es-module-shims.js, line 1)
```
```
[Error] TypeError: Module specifier, '@patternfly/pfe-band' does not start with "/", "./", or "../". Referenced from http://localhost:8080/components/dialog/demo/rh-dialog.js
	promiseReactionJob
[Error] Error: Unable to resolve specifier '@patternfly/pfe-core/decorators/observed.js' imported from http://localhost:8080/components/dialog/rh-dialog.js
	reportError (es-module-shims.js:1:24092)
	throwError (es-module-shims.js:1:1548)
	promiseReactionJob
```



Chrome seems to load `rh-alert`, but fails on what is paritially the same error on `rh-dialog` we see in Safari:
```
Uncaught TypeError: Failed to resolve module specifier "@patternfly/pfe-core/decorators/observed.js". Relative references must start with either "/", "./", or "../".
```

Similar errors were observed when looking at Patternfly demos through `docs:watch`:

Chrome loading `localhost/components/button/demo`
```
An import map is added after module script load was triggered.
Uncaught TypeError: Failed to resolve module specifier "@patternfly/pfe-band". Relative references must start with either "/", "./", or "../".
```

Safari loading `localhost/components/button/demo`
```
[Error] TypeError: Module specifier, '@patternfly/pfe-band' does not start with "/", "./", or "../". Referenced from http://localhost:8081/components/button/demo/pfe-button.js
	promiseReactionJob
^^ Module TypeError above is polyfilled and can be ignored ^^
```

Firefox loading `localhost/components/button/demo`
```
Uncaught TypeError: Error resolving module specifier “@patternfly/pfe-band”. Relative module specifiers must start with “./”, “../” or “/”.
^^ Module TypeError above is polyfilled and can be ignored ^^
```

zeroedin commented:
A separate issue appears to be no demo pages being generated for footer at all:

```
    → footer/index.html              docs/components/footer.njk                 33.2kB    9.2ms
    ↘ footer-block/                  --                                             --       --
      → api/index.html               docs/components/components.md              15.2kB    0.0ms
    ↘ footer-copyright/              --                                             --       --
      → api/index.html               docs/components/components.md              15.2kB    0.0ms
    ↘ footer-links/                  --                                             --       --
      → api/index.html               docs/components/components.md              15.2kB    0.0ms
    ↘ footer-social-link/            --                                             --       --
      → api/index.html               docs/components/components.md              15.2kB    0.0ms
```

brianferry commented:
Looks like a fix for the first issue might be moving

 <script type="importmap">{{ importMap | dump | safe }}</script>

above

{%- for href in preloads -%}
    <link rel="modulepreload" href="{{ href }}"/>
{%- endfor -%}

in docs/components/demo.html

bennypowers commented:
Very helpful notes, thanks

- Added some (dumb, conventional) redirect rules for lightdom css
- fixed the _demo files_ in rhds to point to the right files. or rather, _unfixed_ the mangled paths that were a workaround for the previous behaviour

I'm going to work on
- unifying the urls for demos (so that ux-dot and dev-server use the same pathnames)
- multiple demos in dev-server (with same urls as ux-dot, and a nav ui)
then i'll push those changes up and ping you for review

bennypowers commented:
ok these changes give us:
- congruent urls (so `http://localhost:8000/components/accordion/demo/` is equivalent to `https://ux.redhat.com/components/accordion/demo/`)
- multiple demos grouped by primary element
  - `/components/accordion/demo/`
  - `/components/accordion/demo/pfe-accordion-panel/`  
  - `/components/accordion/demo/accessbility/`

Once this is in place, we can use the demo links on ux-dot as part of the IA to display livecoding playgrounds

bennypowers commented on elements/pfe-button/demo/pfe-button.js:
```diff
@@ -40,7 +40,7 @@ for (const button of document.querySelectorAll('pfe-button')) {
 
 (async function() {
-  const PfeIcon = await customElements.whenDefined('pfe-icon');
+  const PfeIcon = await customElements.get('pfe-icon');
```
[whenDefined](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined) resolves with the class when the tag name is defined

brianferry reviewed (approved):
Looks good to me, tested it locally and everything looked good
